src/main/java/seedu/address/logic/commands/ChangePasswordCommand.java

```diff
@@ -0,0 +1,139 @@
+            + "do not match. Please try again\n";
+
+    private final Logger logger = LogsCenter.getLogger(ChangePasswordCommand.class);
+
+    private String oldPassword;
+    private String newPassword;
+    private String confirmationPassword;
+
+    /**
+     * Takes in old password, new password and confirmation password from parser and creates a new
+     * ChangePasswordCommand object
+     * @param oldPassword
+     * @param newPassword
+     * @param confirmPassword
+     */
+    public ChangePasswordCommand(String oldPassword, String newPassword, String confirmPassword) {
+        this.oldPassword = oldPassword;
+        this.newPassword = newPassword;
+        this.confirmationPassword = confirmPassword;
+    }
+
+    /**
+     * Forward hashes string using SHA256 encryption and returns hashed string
+     * @param argument
+     */
+    private String forwardHash(String argument) {
+        return Hashing.sha256().hashString(argument, StandardCharsets.UTF_8).toString();
+    }
+
+    /**
+     * Forward hashes the user input password and checks if it matches with the encrypted password saved
+     */
+    private boolean isOldPasswordCorrect() {
+        String forwardHashedInputPassword = forwardHash(oldPassword);
+        String forwardHashActualPassword;
+        UserPrefs userPrefs = model.getUserPrefs();
+        forwardHashActualPassword = userPrefs.getAddressBookEncryptedPassword();
+        if (forwardHashActualPassword.equals(forwardHashedInputPassword)) {
+            logger.info("Actual password and input password matches");
+            return true;
+        } else {
+            logger.warning("Actual password and input password do not match");
+            return false;
+        }
+    }
+
+    /**
+     * Takes new input passwords and checks them against one another.
+     */
+    private boolean isNewPasswordInputsSame() {
+        return newPassword.equals(confirmationPassword);
+    }
+
+    /**
+     * Checks if old password and new password are the same
+     */
+    private boolean isOldAndNewPasswordTheSame() {
+        return oldPassword.equals(newPassword);
+    }
+
+    @Override
+    public CommandResult execute() {
+        // Case where old password is incorrect
+        if (!isOldPasswordCorrect()) {
+            logger.warning("Password is incorrect. Note: Default password is 'password' ");
+            return new CommandResult(MESSAGE_PASSWORD_INCORRECT);
+        }
+
+        // Case where new password and confirmation password do not match
+        if (!isNewPasswordInputsSame()) {
+            logger.warning("New password and confirmation password do not match");
+            return new CommandResult(MESSAGE_PASSWORD_CONFIRMATION_INCORRECT);
+        }
+
+        // Case where old and new passwords are the same
+        if (isOldAndNewPasswordTheSame()) {
+            logger.warning("Old password and new password cannot be the same");
+            return new CommandResult(MESSAGE_OLD_NEW_PS_SAME);
```

Comment:
Rather than returning a CommandResult, throwing a CommandException might be more appropriate instead as the command did not pass. The logging will not be needed in this case.

Reply:
That is true. I was wondering why it did not highlight red. Did not know about the CommandException. Thanks :)